projects/dendrites/permutedMNIST/experiments/hyperparameters_figures.py

```diff
@@ -0,0 +1,385 @@
+        hue=hue3,
+        data=df3,
+        palette=pal,
+        bw=sigma,
+        width_viol=0.6,
+        ax=ax3,
+        orient=ort,
+        move=0.2,
+        pointplot=True,
+        alpha=0.65,
+    )
+    l, h = ax3.get_legend_handles_labels()
+    ax3.legend(handles=l[0:8], labels=h[0:8], fontsize="8")
+
+    pt.RainCloud(
+        x=x1,
+        y=y,
+        hue=hue1,
+        data=df1_50,
+        palette=pal,
+        bw=sigma,
+        width_viol=0.6,
+        ax=ax1_50,
+        orient=ort,
+        move=0.2,
+        pointplot=True,
+        alpha=0.65,
+    )
+    l, h = ax1_50.get_legend_handles_labels()
+    labels = h[0:9]
+    ax1_50.legend(
+        handles=l[0:9],
+        labels=labels,
+        fontsize="8",
+    )
+
+    pt.RainCloud(
+        x=x1,
+        y=y,
+        hue=hue2,
+        data=df2_50,
+        palette=pal,
+        bw=sigma,
+        width_viol=0.6,
+        ax=ax2_50,
+        orient=ort,
+        move=0.2,
+        pointplot=True,
+        alpha=0.65,
+    )
+    l, h = ax2_50.get_legend_handles_labels()
+    ax2_50.legend(handles=l[0:8], labels=h[0:8], fontsize="8")
+
+    pt.RainCloud(
+        x=x1,
+        y=y,
+        hue=hue3,
+        data=df3_50,
+        palette=pal,
+        bw=sigma,
+        width_viol=0.6,
+        ax=ax3_50,
+        orient=ort,
+        move=0.2,
+        pointplot=True,
+        alpha=0.65,
+    )
+    l, h = ax3_50.get_legend_handles_labels()
+    ax3_50.legend(handles=l[0:8], labels=h[0:8], fontsize="8")
+
+    ax1.set_xlabel("")
+    ax1.set_ylabel("Mean Accuracy")
+    ax1.set_title("Number of segments")
+    ax1_50.set_xlabel("Tasks", fontsize=16)
+    ax1_50.set_ylabel("Mean Accuracy")
+    ax1_50.xaxis.set_major_locator(ticker.MultipleLocator(10))
+    ax1_50.xaxis.set_major_formatter(ticker.ScalarFormatter())
+    ax2.set_title("Activation density")
+    ax2.set_xlabel("")
+    ax2.set_ylabel("")
+    ax2_50.set_xlabel("Tasks", fontsize=16)
+    ax2_50.set_ylabel("")
+    ax2_50.xaxis.set_major_locator(ticker.MultipleLocator(10))
+    ax2_50.xaxis.set_major_formatter(ticker.ScalarFormatter())
+    ax3.set_xlabel("")
+    ax3.set_ylabel("")
+    ax3.set_title("FF weight density")
+    ax3_50.set_xlabel("Tasks", fontsize=16)
+    ax3_50.set_ylabel("")
+    ax3_50.xaxis.set_major_locator(ticker.MultipleLocator(10))
+    ax3_50.xaxis.set_major_formatter(ticker.ScalarFormatter())
+
+    plt.figtext(-0.01, 0.7, "  10 TASKS", fontsize=14)
+    plt.figtext(-0.01, 0.28, "  50 TASKS", fontsize=14)
+
+    if savefigs:
+        plt.savefig(
+            f"{figs_dir}/hyperparameter_search_panel_along_tasks.png",
+            # bbox_inches="tight",
+        )
+
+
+if __name__ == "__main__":
+
+    savefigs = True
+    figs_dir = "figs/"
+    if savefigs:
+        if not os.path.isdir(f"{figs_dir}"):
+            os.makedirs(f"{figs_dir}")
+
+    experiment_folder = "~/nta/nupic.research/projects/dendrites/permutedMNIST/experiments/data_hyperparameter_search/"
```

Comment:
You shouldn't have hardcoded paths like this in any code that you check in!

Reply:
You can make it relative to the location of the script, such as `../experiments/...`

Reply:
done